core/modules/sql/SqlConnection.cc

```diff
@@ -126,14 +127,29 @@ SqlConnection::~SqlConnection() {
 
 bool
 SqlConnection::connectToDb(SqlErrorObject& errObj) {
-    assert(_connection.get());
-    if (_connection->connected()) return true;
+    assert(_connection != nullptr);
+    static int lastChecked = 0;
+    time_t currentTime = time(0);
+
+    if (_connection->connected()) {
+        int rc = mysql_ping(_connection->getMySql());
+        if (rc == 0) {
+            if (currentTime < lastChecked+60) return true;
+            lastChecked = currentTime;
+            return true;
+        } else {
+            _connection->closeMySqlConn();
+            _connection->connect();
```

Comment:
`_connection->connect();` is called again below, so it will be called twice in this case.